python/dolfinx/fem/element.py

```diff
@@ -160,3 +162,185 @@ def _(e: basix.finite_element.FiniteElement):
+
+        For example, Raviart-Thomas elements will need DOF transformations, as the neighbouring
+        cells may disagree on the orientation of a basis function, and this orientation cannot be
+        corrected for by permuting the DOF numbers on each cell.
+        """
+        return self._cpp_object.needs_dof_transformations
+
+    @property
+    def signature(self) -> str:
+        """String identifying the finite element."""
+        return self._cpp_object.signature
+
+    def T_apply(self, x: npt.NDArray[np.floating], cell_permutations: np.int32, dim: int) -> None:
+        """Transform basis functions from the reference element ordering and orientation to the
+        globally consistent physical element ordering and orientation.
+
+        Args:
+            x: Data to transform (in place). The shape is `(m, n)`, where `m` is the number of
+            dgerees-of-freedom and the storage is row-major.
+            cell_permutations: Permutation data for the cell.
+            dim: Number of columns in `data`.
+
+        Note:
+            Exposed for testing only, expect poor performance.
+        """
+        self._cpp_object.T_apply(x, cell_permutations, dim)
+
+    def Tt_apply(self, x: npt.NDArray[np.floating], cell_permutations: np.int32, dim: int) -> None:
+        """Apply the transpose of the operator applied by T_apply().
+
+        Args:
+            x: Data to transform (in place). The shape is `(m, n)`, where `m` is the number of
+            dgerees-of-freedom and the storage is row-major.
+            cell_permutations: Permutation data for the cell.
+            dim: Number of columns in `data`.
+
+        Note:
+            Exposed for testing only, expect poor performance.
+        """
+        self._cpp_object.Tt_apply(x, cell_permutations, dim)
+
+    def Tt_inv_apply(
+        self, x: npt.NDArray[np.floating], cell_permutations: np.int32, dim: int
+    ) -> None:
+        """Apply the inverse transpose of the operator applied by T_apply().
+
+        Args:
+            x: Data to transform (in place). The shape is `(m, n)`, where `m` is the number of
+            dgerees-of-freedom and the storage is row-major.
+            cell_permutations: Permutation data for the cell.
+            dim: Number of columns in `data`.
+
+        Note:
+            Exposed for testing only, expect poor performance.
+        """
+        self._cpp_object.Tt_apply(x, cell_permutations, dim)
+
+
+def finite_element(
+    cell_type: _cpp.mesh.CellType,
+    ufl_e: ufl.finiteelement,
+    dtype: np.dtype,
+) -> FiniteElement:
+    """Create a DOLFINx element from a basix.ufl element.
+
+    Args:
+        cell_type: Element cell type, see `mesh.CellType`
+        ufl_e: UFL element, holding quadrature rule and other properties of the selected element.
+        dtype: Geometry type of the element.
+    """
+    if np.issubdtype(dtype, np.float32):
+        CppElement = _cpp.fem.FiniteElement_float32
```

Comment:
Let's call this something consistent with the underlying class e.g. FiniteElement_dtype

On a tangential note it would be nice to automate this type of duck typing across the dtype.